src/table_properties.rs

```diff
@@ -0,0 +1,244 @@
+use std::collections::HashMap;
+use std::ffi::{c_char, c_void, CStr, CString};
+use std::marker::PhantomData;
+use std::mem;
+use std::slice;
+
+use libc::{c_int, size_t};
+
+use crate::{ffi, Options};
+
+/// Extension trait for [`Options`] to register table properties collectors
+pub trait TablePropertiesExt {
+    fn add_table_properties_collector_factory<F>(&mut self, factory_fn: F)
```

Comment:
A potential simplification to the API is to replace the trait with an owned `CString` name and `FnMut(TablePropertiesCollectorContext) -> TablePropertiesCollector`. I thought it wasn't that much of an improvement - the trait is more consistent with the rest of `rust-rocksdb`. Let me know if you disagree!

Reply:
PS. This should be:

```suggestion
    fn add_table_properties_collector_factory<F>(&mut self, factory: F)
```

Missed cleaning that up.